Examples/Streaming/README.md

```diff
@@ -0,0 +1,217 @@
+
+To build & archive the package, type the following commands.
+
+```bash
+swift package archive --allow-network-connections docker
+```
+
+If there is no error, there is a ZIP file ready to deploy. 
+The ZIP file is located at `.build/plugins/AWSLambdaPackager/outputs/AWSLambdaPackager/StreamingNumbers/StreamingNumbers.zip`
+
+## Deploy with the AWS CLI
+
+Here is how to deploy using the `aws` command line.
+
+### Step 1: Create the function 
+```bash
+AWS_ACCOUNT_ID=012345678901
+aws lambda create-function \
+--function-name StreamingNumbers \
+--zip-file fileb://.build/plugins/AWSLambdaPackager/outputs/AWSLambdaPackager/StreamingNumbers/StreamingNumbers.zip \
+--runtime provided.al2 \
+--handler provided  \
+--architectures arm64 \
+--role arn:aws:iam::${AWS_ACCOUNT_ID}:role/lambda_basic_execution \
+--timeout 15
+```
+
+> [!IMPORTANT] 
+> The timeout value must be bigger than the time it takes for your function to stream its output. Otherwise, the Lambda control plane will terminate the execution environment before your code has a chance to finish writing the stream. Here, teh sample function stream responses during 10 seconds and we set the timeout for 15 seconds.
+
+The `--architectures` flag is only required when you build the binary on an Apple Silicon machine (Apple M1 or more recent). It defaults to `x64`.
+
+Be sure to set `AWS_ACCOUNT_ID` with your actual AWS account ID (for example: 012345678901).
+
+### Step2: Give permission to invoke that function through an URL
+
+Anyone with a valid signature from your AWS account will have permission to invoke the function through its URL.
+
+```bash
+aws lambda add-permission \
+  --function-name StreamingNumbers \
+  --action lambda:InvokeFunctionUrl \
+  --principal $AWS_ACCOUNT_ID \
+  --function-url-auth-type AWS_IAM \
+  --statement-id allowURL
+```  
+
+Be sure to replace <YOUR_ACCOUNT_ID> with your actual AWS account ID (for example: 012345678901).
+
+### Step3: Create the URL 
+
+This creates [an URL with IAM authentication](https://docs.aws.amazon.com/lambda/latest/dg/urls-auth.html). Only calls with a valid signature will be authorized.
+
+```bash
+aws lambda create-function-url-config \
+  --function-name StreamingNumbers \
+  --auth-type AWS_IAM \
+  --invoke-mode RESPONSE_STREAM 
+```
+This calls return various information, including the URL to invoke your function.
+
+```json
+{
+    "FunctionUrl": "https://ul3nf4dogmgyr7ffl5r5rs22640fwocc.lambda-url.us-east-1.on.aws/",
+    "FunctionArn": "arn:aws:lambda:us-east-1:012345678901:function:StreamingNumbers",
+    "AuthType": "AWS_IAM",
+    "CreationTime": "2024-10-22T07:57:23.112599Z",
+    "InvokeMode": "RESPONSE_STREAM"
+}
+```
+
+### Invoke your Lambda function
+
+To invoke the Lambda function, use `curl` with the AWS Sigv4 option to generate the signature.
+```bash
+URL=https://ul3nf4dogmgyr7ffl5r5rs22640fwocc.lambda-url.us-east-1.on.aws/
+REGION=us-east-1
+ACCESS_KEY=AK...
+SECRET_KEY=...
+AWS_SESSION_TOKEN=...
```

Comment:
Where does the session token come from? Would be good to show how to create it, or link to a doc that shows it

Reply:
added a chapter about AWS credentials